### PR TITLE
wharf-core logging & configs

### PR DESCRIPTION
**Pull request by applejag**
- \[x] I've added a new note in the `CHANGELOG.md` file, according to docs:
  https://iver-wharf.github.io/#/development/changelogs/writing-changelogs

## Summary

- Basic logging & config via wharf-core
- Swaggo/swag v1.7.1
- Using wharf-core logging

## Motivation

Keeping the projects up to date with our own packages

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	// (v3.0.0), the environment variable BIND_ADDRESS, which was added in
+	// v2.0.0, will also set this value.
+	//
+	// Added in v1.3.0
+	BindAddress string
+}
+
+// CORSConfig holds settings for the HTTP server's CORS settings.
+type CORSConfig struct {
+	// AllowAllOrigins enables all
```
Comment (Alexamakans) [inline config.go]: It seems like part of the comment has gotten chopped here.

Comment (applejag) [reply config.go]: Applying same suggestion as from https://github.com/iver-wharf/wharf-provider-gitlab/pull/23#pullrequestreview-736934503

```suggestion
	// AllowAllOrigins enables CORS and allows all hostnames and URLs in the
	// HTTP request origins when set to true. Practically speaking, this
	// results in the HTTP header "Access-Control-Allow-Origin" set to "*".
```

**Review by Alexamakans — APPROVED** [review]

File: internal/azureapi/client.go [hunk 2]
```diff
@@ -8,17 +8,20 @@ import (
...
-	Context  *gin.Context
-	BaseURL  string
+	Context *gin.Context
+	BaseURL string
 	// BaseURLParsed is the result of url.Parse(BaseURL)
 	BaseURLParsed *url.URL
-	UserName string
-	Token    string
+	UserName      string
+	Token         string
```
Comment (fredx30) [inline internal/azureapi/client.go:24]: nit: Looks like redundant spacing

File: main.go [hunk 3]
```diff
@@ -22,36 +28,59 @@ import (
...
 
-	r := gin.Default()
+	if config.CA.CertsFile != "" {
+		client, err := httputils.NewClientWithCerts(config.CA.CertsFile)
+		if err != nil {
+			log.Error().WithError(err).Message("Failed to get net/http.Client with certs.")
+			os.Exit(1)
+		}
+		http.DefaultClient = client
+	}
```
Comment (fredx30) [inline main.go:55]: The need for certificates related changes does not seem warranted by whats in the description for the PR. Can you explain?

File: internal/azureapi/client.go (hunk 2 above)
Comment (applejag) [reply internal/azureapi/client.go:24]: Automatically applied by `gofmt`/`goimports`. It's to align it with the `*url.URL` on the line above

File: main.go (hunk 3 above)
Comment (applejag) [reply main.go:55]: Forgot to add a note about that. As I copied over the config from the other APIs to reduce work, I noticed this repo didn't have the cert support that the other repos do.

It's out of scope for this PR, and I'll consider moving it to a different PR.

Comment (fredx30) [reply main.go:55]: Good enough for me! You can probably merge it here. Just add a small note in the changelog.

**Review by fredx30 — APPROVED** [review]

**Pull request merged**